### ISSUE TITLE
Restart action from admin dashboard fails with 'database is locked'

### ISSUE THREAD

**Issue by Terrance** [jellyfin/jellyfin#13058]
### This issue respects the following points:

- [x] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [x] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [x] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [x] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [x] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

Triggering a Jellyfin server restart in the UI (via the admin dashboard) leads to a crash where the database is locked, presumably because something hasn't released it during the restart.

### Reproduction steps

1. Menu > _Administration_ > _Dashboard_
2. _Restart_ under the server info, and _Restart_ again to confirm

### What is the current _bug_ behavior?

Server shuts down, begins starting up again, but crashes:

```
[FTL] Error while starting server
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 8: 'attempt to write a readonly database'.
```

Starting the server again at this point (by starting the systemd service) works without any further issues.

### What is the expected _correct_ behavior?

Server shuts down, starts up again normally.

### Jellyfin Server version

10.10.0+

### Specify the build version

Release

### Environment

```markdown
- OS: Arch Linux
- Linux Kernel: 6.11.8-arch1-2
- Virtualization: none
- Clients: Browser
- Browser: Firefox 133.0b9
- FFmpeg Version: 7.1 (system), 7.0.2 (Jellyfin embedded)
- Playback Method: n/a
- Hardware Acceleration: none
- GPU Model: none
- Plugins: ListenBrainz (via https://repo.xkrivo.net/jellyfin/manifest.json)
- Reverse Proxy: nginx
- Base URL: /
- Networking: host
- Storage: local
```

### Jellyfin logs

```shell
[11:07:07] [INF] Sending shutdown notifications
[11:07:07] [INF] WS 172.16.2.1 closed
[11:07:07] [INF] Stopping NAT discovery
[11:07:07] [INF] Disposing SsdpDevicePublisher
[11:07:07] [INF] Disposing PlayToManager
[11:07:07] [INF] Running query planner optimizations in the database... This might take a while
[11:07:07] [INF] SsdpCommunicationsServer disposing _BroadcastListenSocket
[11:07:07] [INF] SsdpCommunicationsServer Disposing 1 sendSockets
[11:07:07] [INF] SsdpCommunicationsServer disposing sendSocket from 172.16.1.1
[11:07:07] [INF] Stopping directory watching for path ...
[11:07:08] [INF] Disposing CoreAppHost
[11:07:08] [INF] Disposing MusicBrainzArtistProvider
[11:07:08] [INF] Disposing MusicBrainzAlbumProvider
[11:07:08] [INF] Disposing PluginManager
[11:07:08] [INF] Setting cache path: /var/cache/jellyfin
[11:07:08] [INF] Loading assemblies
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Bookshelf, Version=12.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Bookshelf_12.0.0.0/Jellyfin.Plugin.Bookshelf.dll
[11:07:08] [INF] Loaded assembly SharpCompress, Version=0.38.0.0, Culture=neutral, PublicKeyToken=afb0a02973931d96 from /var/lib/jellyfin/plugins/Bookshelf_12.0.0.0/SharpCompress.dll
[11:07:08] [INF] Loaded assembly ZstdSharp, Version=0.8.1.0, Culture=neutral, PublicKeyToken=8d151af33a4ad5cf from /var/lib/jellyfin/plugins/Bookshelf_12.0.0.0/ZstdSharp.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Dlna.Model, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/DLNA_5.0.0.0/Jellyfin.Plugin.Dlna.Model.dll
[11:07:08] [INF] Loaded assembly Rssdp, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/DLNA_5.0.0.0/Rssdp.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Dlna.Playback, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/DLNA_5.0.0.0/Jellyfin.Plugin.Dlna.Playback.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Dlna, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/DLNA_5.0.0.0/Jellyfin.Plugin.Dlna.dll
[11:07:08] [INF] Skipping disabled plugin 8.1.0.0 of Last.fm
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.ListenBrainz.Api, Version=5.0.0.2, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/ListenBrainz_5.0.0.2/Jellyfin.Plugin.ListenBrainz.Api.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.ListenBrainz, Version=5.0.0.2, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/ListenBrainz_5.0.0.2/Jellyfin.Plugin.ListenBrainz.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.ListenBrainz.Http, Version=5.0.0.2, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/ListenBrainz_5.0.0.2/Jellyfin.Plugin.ListenBrainz.Http.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.ListenBrainz.MusicBrainzApi, Version=5.0.0.2, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/ListenBrainz_5.0.0.2/Jellyfin.Plugin.ListenBrainz.MusicBrainzApi.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.ListenBrainz.Common, Version=5.0.0.2, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/ListenBrainz_5.0.0.2/Jellyfin.Plugin.ListenBrainz.Common.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.OpenSubtitles, Version=20.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Open Subtitles_20.0.0.0/Jellyfin.Plugin.OpenSubtitles.dll
[11:07:08] [INF] Loaded assembly SQLitePCL.pretty, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Playback Reporting_15.0.0.0/SQLitePCL.pretty.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.PlaybackReporting, Version=15.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Playback Reporting_15.0.0.0/Jellyfin.Plugin.PlaybackReporting.dll
[11:07:08] [INF] Loaded assembly System.IO.Packaging, Version=4.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a from /var/lib/jellyfin/plugins/Reports_17.0.0.0/System.IO.Packaging.dll
[11:07:08] [INF] Loaded assembly SixLabors.Fonts, Version=1.0.0.0, Culture=neutral, PublicKeyToken=d998eea7b14cab13 from /var/lib/jellyfin/plugins/Reports_17.0.0.0/SixLabors.Fonts.dll
[11:07:08] [INF] Loaded assembly ExcelNumberFormat, Version=1.1.0.0, Culture=neutral, PublicKeyToken=23c6f5d73be07eca from /var/lib/jellyfin/plugins/Reports_17.0.0.0/ExcelNumberFormat.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Reports, Version=17.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Reports_17.0.0.0/Jellyfin.Plugin.Reports.dll
[11:07:08] [INF] Loaded assembly ClosedXML, Version=0.97.0.0, Culture=neutral, PublicKeyToken=fd1eb21b62ae805b from /var/lib/jellyfin/plugins/Reports_17.0.0.0/ClosedXML.dll
[11:07:08] [INF] Loaded assembly DocumentFormat.OpenXml, Version=2.16.0.0, Culture=neutral, PublicKeyToken=8fb06cb64d019a17 from /var/lib/jellyfin/plugins/Reports_17.0.0.0/DocumentFormat.OpenXml.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.TMDbBoxSets, Version=11.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/TMDb Box Sets_11.0.0.0/Jellyfin.Plugin.TMDbBoxSets.dll
[11:07:08] [INF] Loaded assembly Trakt, Version=26.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Trakt_26.0.0.0/Trakt.dll
[11:07:08] [INF] Loaded assembly Handlebars, Version=2.1.6.0, Culture=neutral, PublicKeyToken=22225d0bf33cd661 from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/Handlebars.dll
[11:07:08] [INF] Loaded assembly BouncyCastle.Cryptography, Version=2.0.0.0, Culture=neutral, PublicKeyToken=072edcf4a5328938 from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/BouncyCastle.Cryptography.dll
[11:07:08] [INF] Loaded assembly MailKit, Version=4.7.0.0, Culture=neutral, PublicKeyToken=4e064fe7c44a8f1b from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/MailKit.dll
[11:07:08] [INF] Loaded assembly MimeKit, Version=4.7.0.0, Culture=neutral, PublicKeyToken=bede1c8a46c66814 from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/MimeKit.dll
[11:07:08] [INF] Loaded assembly MQTTnet.Extensions.ManagedClient, Version=4.3.6.1152, Culture=neutral, PublicKeyToken=fdb7629f2e364a63 from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/MQTTnet.Extensions.ManagedClient.dll
[11:07:08] [INF] Loaded assembly Jellyfin.Plugin.Webhook, Version=15.0.0.0, Culture=neutral, PublicKeyToken=null from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/Jellyfin.Plugin.Webhook.dll
[11:07:08] [INF] Loaded assembly MQTTnet, Version=4.3.6.1152, Culture=neutral, PublicKeyToken=fdb7629f2e364a63 from /var/lib/jellyfin/plugins/Webhook_15.0.0.0/MQTTnet.dll
[11:07:08] [INF] Defined LAN subnets: ["172.16.0.0/16", ...]
[11:07:08] [INF] Defined LAN exclusions: []
[11:07:08] [INF] Used LAN subnets: ["172.16.0.0/16", ...]
[11:07:08] [INF] Filtered interface addresses: ["127.0.0.1", "172.16.1.1", ...]
[11:07:08] [INF] Bind Addresses ["::"]
[11:07:08] [INF] Remote IP filter is Allowlist
[11:07:08] [INF] Filtered subnets: []
[11:07:08] [FTL] Error while starting server
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 8: 'attempt to write a readonly database'.
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Emby.Server.Implementations.Data.SqliteExtensions.Execute(SqliteConnection sqliteConnection, String commandText)
   at Emby.Server.Implementations.Data.ManagedConnection.Execute(String commandText)
   at Emby.Server.Implementations.Data.BaseSqliteRepository.Initialize()
   at Emby.Server.Implementations.Data.SqliteItemRepository.Initialize()
   at Emby.Server.Implementations.ApplicationHost.InitializeServices()
   at Jellyfin.Server.Program.StartServer(IServerApplicationPaths appPaths, StartupOptions options, IConfiguration startupConfig)
[11:07:08] [INF] Running query planner optimizations in the database... This might take a while
[11:07:08] [INF] Disposing CoreAppHost
[11:07:08] [INF] Disposing PluginManager
```

### FFmpeg logs

```shell

```

### Additional information

I can see a handful of other 'database is locked', though I don't think any apply specifically to this flow.

First noticed in Jellyfin 10.10.0, still present in 10.10.1 (though I only really just started using this option after noticing plugins effectively require two restarts, one to trigger the update and another to load it).

**Comment by sjorge** [comment]
It's an issue with the PlaybackReporting Plugin.

Removing it will fix the restart button (and a whole lot of random weird database is locked errors)

**Comment by Terrance** [comment]
Apologies, I hadn't factored Jellyfin-maintained plugins into the list; yes, I do also have that installed, though the logs don't make it clear that that's the source of the database issues!

I seem to be unable to disable it in the UI though -- after a shutdown and startup again, a previous version seems to active instead, and once startup is complete Jellyfin redownloads the latest version as an update, which activates again on the next startup:

![Image](https://github.com/user-attachments/assets/d69a7905-d8d8-481d-8616-f21f54201456)

```
[INF] Core startup complete
[INF] Startup complete 0:00:17.054741
[INF] Plugin updated: Playback Reporting 15.0.0.0
[INF] App needs to be restarted.
[INF] Plugin updated: Playback Reporting 15.0.0.0
[INF] App needs to be restarted.
[INF] Update Plugins Completed after 0 minute(s) and 3 seconds
```

Guessing I'll need to dig into the data directories to manually remove the plugin, but I can hold off for now if there's anything useful I can help to debug here -- presumably this isn't the expected effect of disabling a plugin; I'd expect any lingering older versions to be disabled too.

> It's an issue with the PlaybackReporting Plugin.

As [the repo for that appears to be in the Jellyfin org](https://github.com/jellyfin/jellyfin-plugin-playbackreporting), can someone migrate this issue over there?  (Or else I can raise a new one.)

**Comment by sjorge** [comment]
Yeah you're probably going to have to remove the old version from disk, for some reason on update it doesn't remove the old versions. So it just keeps updating every startup.

Probably OK to close this and open a new issue on the plugin repo. Not sure issues can be moved between repos.

**Comment by Terrance** [comment]
Having moved all instances of the Playback Reporting plugin out of the way, the server does indeed seem to restart correctly.

> Yeah you're probably going to have to remove the old version from disk, for some reason on update it doesn't remove the old versions.

Looking in `~jellyfin/config/system.xml` I can see the [server option `RemoveOldPlugins`](https://github.com/jellyfin/jellyfin/blob/d1d5ea9c80d254aba9d749f1fad5ace5b31579ab/MediaBrowser.Model/Configuration/ServerConfiguration.cs#L249) (= `false`), though I don't see a UI option which corresponds to it.  Seems like that would probably help here, though I'm unsure if that's a "supported" option, or not intended for regular use seeing as it's not offered via the UI.

> Not sure issues can be moved between repos.

Fwiw I believe they can [transferred within a GitHub org](https://docs.github.com/en/issues/tracking-your-work-with-issues/administering-issues/transferring-an-issue-to-another-repository) by its owners (I'm assuming you're not one of them from the lack of badges next to your name), but seeing as I've confirmed things behave without the plugin, I can go ahead and raise a separate issue.

**Comment by Shadowghost** [comment]
You can enabled ´RemoveOldPlugins` without any issues (not sure why we never added a UI option for it but afaik on new installs it's defaulted to `true`).

**Comment by Bond-009** [comment]
Closing as this this is a plugin issue

**Comment by Terrance** [comment]
I had left this open for https://github.com/jellyfin/jellyfin/issues/13058#issuecomment-2481398599, as that seems to happen with any plugins when multiple versions exist, though after manually removing the older versions I can now disable the single installed version as expected.

**Comment by Bond-009** [comment]
Old versions should be removed by default on newer installs.
If you still think that it's an issue I recommend opening a new issue for it instead of keeping this open.

**Comment by xDaveN** [comment]
Just ran into this. Having any version of the playback reporting plugin installed bricks my ability to restart jellyfin from the dashboard - with the same db error.

Cleaned up old versions manually and set RemoveOldPlugins to true, just in case. Oh well, guess this really is plugin specific.
